plugins/woocommerce/client/admin/client/task-lists/fills/Marketing/TaskPromo.tsx

```diff
@@ -0,0 +1,70 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Button, Card, CardBody, CardHeader } from '@wordpress/components';
+import { Text } from '@woocommerce/experimental';
+
+/**
+ * Internal dependencies
+ */
+import './TaskPromo.scss';
+import { WC_ASSET_URL } from '~/utils/admin-settings';
+
+export type TaskPromoProps = {
+	title?: string;
+	iconSrc?: string;
+	iconAlt?: string;
+	name?: string;
+	text?: string;
+	buttonHref?: string;
+	buttonText?: string;
+	onButtonClick?: () => void;
+};
+
+export const TaskPromo: React.FC< TaskPromoProps > = ( {
+	title = '',
+	iconSrc = `${ WC_ASSET_URL }images/woo-app-icon.svg`,
+	iconAlt = __( 'Woo icon', 'woocommerce' ),
+	name = __( 'WooCommerce Marketplace', 'woocommerce' ),
+	text = '',
+	buttonHref = '',
+	buttonText = '',
+	onButtonClick,
+} ) => {
+	return (
+		<Card className="woocommerce-task-card woocommerce-task-promo">
+			{ title && (
+				<CardHeader>
+					<Text
+						variant="title.small"
+						as="h2"
+						className="woocommerce-task-card__title"
+					>
+						{ title }
+					</Text>
+				</CardHeader>
+			) }
+			<CardBody>
+				<div className="woocommerce-plugin-list__plugin-logo">
+					<img src={ iconSrc as string } alt={ iconAlt as string } />
```

Comment:
I believe we have an opportunity to be more concise with our types here. 

We accept `iconSrc` and `iconAlt` as optional props. Because they are optional, we need to assure TypeScript that this a string an not a null with `as string`. 

How about we pass the `iconSrc` to `TaskPromo`? So that we are sure it is a string and it's not optional. 

```suggestion
					<img src={ iconSrc } alt={ iconAlt } />
```

In `client/admin/client/task-lists/fills/Marketing/index.tsx`
```tsx
<TaskPromo
   // other props
   iconSrc={ `${ WC_ASSET_URL }images/woo-app-icon.svg` }
   iconAlt={ __( 'Woo icon', 'woocommerce' ) }
/> 
```

Reply:
TS doesn't like it when you try to use an optional prop with a default value in an attribute – it complains `Type string | undefined is not assignable to type string | undefined`. As a node, such a prop is fine, for example `name`, which we define in exactly the same way as `iconAlt`. 

None of the alternative type solutions suggested by Perplexity or Copilot work for me, so in https://github.com/woocommerce/woocommerce/pull/55605/commits/ddda7fe121d136b6dd2417fc2ff0e9533f8faa2c I've introduced what feels like an otherwise unnecessary condition to silence the error.